app/app/upload-non-cr3/page.tsx

```diff
@@ -0,0 +1,257 @@
+"use client";
+import { useState, useCallback } from "react";
+import Link from "next/link";
+import AppBreadCrumb from "@/components/AppBreadCrumb";
+import Alert from "react-bootstrap/Alert";
+import Button from "react-bootstrap/Button";
+import Card from "react-bootstrap/Card";
+import Col from "react-bootstrap/Col";
+import Form from "react-bootstrap/Form";
+import Row from "react-bootstrap/Row";
+import Spinner from "react-bootstrap/Spinner";
+import Table from "react-bootstrap/Table";
+import Papa, { ParseResult } from "papaparse";
+import { z, ZodError } from "zod";
+import AlignedLabel from "@/components/AlignedLabel";
+import { useMutation } from "@/utils/graphql";
+import { INSERT_NON_CR3 } from "@/queries/nonCr3s";
+import {
+  NonCr3Upload,
+  nonCr3UploadSchema,
+  NonCr3ValidationError,
+} from "@/types/nonCr3";
+import {
+  FaFileCsv,
+  FaCircleCheck,
+  FaCircleInfo,
+  FaTriangleExclamation,
+} from "react-icons/fa6";
+
+const MAX_ERRORS_TO_DISPLAY = 50;
+
+export default function UploadNonCr3() {
+  const [parsing, setParsing] = useState(false);
+  const [data, setData] = useState<NonCr3Upload[] | null>(null);
+  const [validationErrors, setValidationErrors] = useState<
+    NonCr3ValidationError[] | null
+  >();
+  const [uploadError, setUploadError] = useState(false);
+  const [success, setSuccess] = useState(false);
+  const { mutate, loading: isMutating } = useMutation(INSERT_NON_CR3);
+
+  const onSelectFile = useCallback(
+    (file: File) => {
+      Papa.parse<NonCr3Upload>(file, {
```

Comment:
the old version of this feature used [react-csv-reader](https://github.com/nzambello/react-csv-reader), which is just a wrapper around `papaparse`.